maintenance_request_ifs_check/models/maintenance.py

```diff
@@ -0,0 +1,40 @@
+from odoo import api, models, fields, _
+from odoo.exceptions import ValidationError
+
+
+class MaintenanceStage(models.Model):
+    _inherit = 'maintenance.stage'
+
+    ifs_checked = fields.Boolean(
+        'IFS checked',
+        default=False)
+
+
+class MaintenanceRequest(models.Model):
+    _inherit = 'maintenance.request'
+
+    ifs_check_repaired_correctly = fields.Boolean(
+        string='Equipment repaired correctly')
+    ifs_check_free_strange_elements = fields.Boolean(
+        string='Free strange elements')
+    ifs_check_collected_location = fields.Boolean(
+        string='Collected location')
+    ifs_check_cleaning_area = fields.Boolean(
+        string='Cleaning the area')
+    ifs_checked = fields.Boolean(
+        string='IFS checked',
+        related='stage_id.ifs_checked',
+        store=True,
+        readonly=True)
+
+    @api.constrains('stage_id')
+    def _check_ifs_checked(self):
+        for req in self:
+            if req.stage_id.ifs_checked\
+                    and (not req.ifs_check_repaired_correctly
+                         or not req.ifs_check_free_strange_elements
+                         or not req.ifs_check_collected_location
+                         or not req.ifs_check_cleaning_area):
```

Comment:
For this piece of code, creating and calling for a `_ifs_checked_ok()` method (or even a computed field) should be clearer, and should be a good technique for further inheritance (e.g. for upcoming addons with new `ifs_*` fields).

Reply:
I understand. I'll think about use a computed field or not.